Remove display list item extent tracking

heycam opened the pull request:
#### d05bc644486e3c73a13c5099665320946dc65857
<pre>
Remove display list item extent tracking
<a href="https://bugs.webkit.org/show_bug.cgi?id=241793">https://bugs.webkit.org/show_bug.cgi?id=241793</a>

Reviewed by Simon Fraser.

Extent tracking is not used for remote display list recording. It is
currently used for in-memory display list repainting, but the only thing
using that is the GlyphDisplayListCache, and it&apos;s not important there.

* Source/WebCore/platform/graphics/DecomposedGlyphs.cpp:
(WebCore::DecomposedGlyphs::create):
(WebCore::DecomposedGlyphs::DecomposedGlyphs):
* Source/WebCore/platform/graphics/DecomposedGlyphs.h:
(WebCore::DecomposedGlyphs::positionedGlyphs const):
(WebCore::DecomposedGlyphs::bounds const): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayList.cpp:
(WebCore::DisplayList::DisplayList::setTracksDrawingItemExtents): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayList.h:
(WebCore::DisplayList::DisplayList::resourceHeap const):
(WebCore::DisplayList::DisplayList::tracksDrawingItemExtents const): Deleted.
(WebCore::DisplayList::DisplayList::addDrawingItemExtent): Deleted.
(): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.cpp:
(WebCore::DisplayList::DrawGlyphs::DrawGlyphs):
(WebCore::DisplayList::m_bounds): Deleted.
(WebCore::DisplayList::DrawLine::localBounds const): Deleted.
(WebCore::DisplayList::DrawLinesForText::localBounds const): Deleted.
(WebCore::DisplayList::DrawDotsForDocumentMarker::localBounds const): Deleted.
(WebCore::DisplayList::DrawFocusRingPath::localBounds const): Deleted.
(WebCore::DisplayList::DrawFocusRingRects::localBounds const): Deleted.
(WebCore::DisplayList::StrokeRect::localBounds const): Deleted.
(WebCore::DisplayList::StrokePath::localBounds const): Deleted.
(WebCore::DisplayList::StrokeEllipse::localBounds const): Deleted.
(WebCore::DisplayList::StrokeLine::localBounds const): Deleted.
(WebCore::DisplayList::StrokeArc::localBounds const): Deleted.
(WebCore::DisplayList::StrokeQuadCurve::localBounds const): Deleted.
(WebCore::DisplayList::StrokeBezierCurve::localBounds const): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
(WebCore::DisplayList::DrawGlyphs::encode const):
(WebCore::DisplayList::DrawGlyphs::decode):
(WebCore::DisplayList::DrawDecomposedGlyphs::DrawDecomposedGlyphs):
(WebCore::DisplayList::DrawFilteredImageBuffer::globalBounds const): Deleted.
(WebCore::DisplayList::DrawFilteredImageBuffer::localBounds const): Deleted.
(WebCore::DisplayList::DrawGlyphs::globalBounds const): Deleted.
(WebCore::DisplayList::DrawGlyphs::localBounds const): Deleted.
(WebCore::DisplayList::DrawDecomposedGlyphs::globalBounds const): Deleted.
(WebCore::DisplayList::DrawDecomposedGlyphs::localBounds const): Deleted.
(WebCore::DisplayList::DrawImageBuffer::globalBounds const): Deleted.
(WebCore::DisplayList::DrawImageBuffer::localBounds const): Deleted.
(WebCore::DisplayList::DrawNativeImage::globalBounds const): Deleted.
(WebCore::DisplayList::DrawNativeImage::localBounds const): Deleted.
(WebCore::DisplayList::DrawSystemImage::globalBounds const): Deleted.
(WebCore::DisplayList::DrawSystemImage::localBounds const): Deleted.
(WebCore::DisplayList::DrawPattern::globalBounds const): Deleted.
(WebCore::DisplayList::DrawPattern::localBounds const): Deleted.
(WebCore::DisplayList::BeginTransparencyLayer::localBounds const): Deleted.
(WebCore::DisplayList::BeginTransparencyLayer::globalBounds const): Deleted.
(WebCore::DisplayList::EndTransparencyLayer::localBounds const): Deleted.
(WebCore::DisplayList::EndTransparencyLayer::globalBounds const): Deleted.
(WebCore::DisplayList::DrawRect::globalBounds const): Deleted.
(WebCore::DisplayList::DrawRect::localBounds const): Deleted.
(WebCore::DisplayList::DrawLine::globalBounds const): Deleted.
(WebCore::DisplayList::DrawLinesForText::globalBounds const): Deleted.
(WebCore::DisplayList::DrawDotsForDocumentMarker::globalBounds const): Deleted.
(WebCore::DisplayList::DrawEllipse::globalBounds const): Deleted.
(WebCore::DisplayList::DrawEllipse::localBounds const): Deleted.
(WebCore::DisplayList::DrawPath::globalBounds const): Deleted.
(WebCore::DisplayList::DrawPath::localBounds const): Deleted.
(WebCore::DisplayList::DrawFocusRingPath::globalBounds const): Deleted.
(WebCore::DisplayList::DrawFocusRingRects::globalBounds const): Deleted.
(WebCore::DisplayList::FillRect::globalBounds const): Deleted.
(WebCore::DisplayList::FillRect::localBounds const): Deleted.
(WebCore::DisplayList::FillRectWithColor::globalBounds const): Deleted.
(WebCore::DisplayList::FillRectWithColor::localBounds const): Deleted.
(WebCore::DisplayList::FillRectWithGradient::globalBounds const): Deleted.
(WebCore::DisplayList::FillRectWithGradient::localBounds const): Deleted.
(WebCore::DisplayList::FillCompositedRect::globalBounds const): Deleted.
(WebCore::DisplayList::FillCompositedRect::localBounds const): Deleted.
(WebCore::DisplayList::FillRoundedRect::globalBounds const): Deleted.
(WebCore::DisplayList::FillRoundedRect::localBounds const): Deleted.
(WebCore::DisplayList::FillRectWithRoundedHole::globalBounds const): Deleted.
(WebCore::DisplayList::FillRectWithRoundedHole::localBounds const): Deleted.
(WebCore::DisplayList::FillLine::globalBounds const): Deleted.
(WebCore::DisplayList::FillLine::localBounds const): Deleted.
(WebCore::DisplayList::FillArc::globalBounds const): Deleted.
(WebCore::DisplayList::FillArc::localBounds const): Deleted.
(WebCore::DisplayList::FillQuadCurve::globalBounds const): Deleted.
(WebCore::DisplayList::FillQuadCurve::localBounds const): Deleted.
(WebCore::DisplayList::FillBezierCurve::globalBounds const): Deleted.
(WebCore::DisplayList::FillBezierCurve::localBounds const): Deleted.
(WebCore::DisplayList::FillPath::globalBounds const): Deleted.
(WebCore::DisplayList::FillPath::localBounds const): Deleted.
(WebCore::DisplayList::FillEllipse::globalBounds const): Deleted.
(WebCore::DisplayList::FillEllipse::localBounds const): Deleted.
(WebCore::DisplayList::PaintFrameForMedia::localBounds const): Deleted.
(WebCore::DisplayList::PaintFrameForMedia::globalBounds const): Deleted.
(WebCore::DisplayList::StrokeRect::globalBounds const): Deleted.
(WebCore::DisplayList::StrokeLine::globalBounds const): Deleted.
(WebCore::DisplayList::StrokeArc::globalBounds const): Deleted.
(WebCore::DisplayList::StrokeQuadCurve::globalBounds const): Deleted.
(WebCore::DisplayList::StrokeBezierCurve::globalBounds const): Deleted.
(WebCore::DisplayList::StrokePath::globalBounds const): Deleted.
(WebCore::DisplayList::StrokeEllipse::globalBounds const): Deleted.
(WebCore::DisplayList::ClearRect::globalBounds const): Deleted.
(WebCore::DisplayList::ClearRect::localBounds const): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.cpp:
(WebCore::DisplayList::Recorder::drawGlyphsAndCacheResources):
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.cpp:
(WebCore::DisplayList::RecorderImpl::recordDrawDecomposedGlyphs):
(WebCore::DisplayList::RecorderImpl::extentFromLocalBounds const): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.h:
(WebCore::DisplayList::RecorderImpl::append):
* Source/WebKit/GPUProcess/graphics/RemoteDisplayListRecorder.cpp:
(WebKit::RemoteDisplayListRecorder::drawDecomposedGlyphs):
(WebKit::RemoteDisplayListRecorder::drawDecomposedGlyphsWithQualifiedIdentifiers):
* Source/WebKit/GPUProcess/graphics/RemoteDisplayListRecorder.h:
* Source/WebKit/GPUProcess/graphics/RemoteDisplayListRecorder.messages.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;DecomposedGlyphs&gt;::encode):
(IPC::ArgumentCoder&lt;DecomposedGlyphs&gt;::decode):
* Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.cpp:
(WebKit::RemoteDisplayListRecorderProxy::recordDrawDecomposedGlyphs):
* LayoutTests/displaylists/extent-includes-shadow-expected.txt: Removed.
* LayoutTests/displaylists/extent-includes-shadow.html: Removed.
* LayoutTests/displaylists/extent-includes-transforms-expected.txt: Removed.
* LayoutTests/displaylists/extent-includes-transforms.html: Removed.
* LayoutTests/displaylists/layer-dispay-list-expected.txt:
* LayoutTests/displaylists/replay-skip-clipped-rect-expected.txt: Removed.
* LayoutTests/displaylists/replay-skip-clipped-rect.html: Removed.
* LayoutTests/fast/text/glyph-display-lists/glyph-display-list-color-expected.txt:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/platform/mac-wk1/displaylists/extent-includes-shadow-expected.txt: Removed.
* LayoutTests/platform/mac-wk1/displaylists/extent-includes-transforms-expected.txt: Removed.

Canonical link: <a href="https://commits.webkit.org/252442@main">https://commits.webkit.org/252442@main</a>
</pre>

smfr reviewed (approved):
There's no code that uses the bounds on any display list object right now; we should probably just remove them entirely.

heycam committed:
Remove display list item extent tracking
https://bugs.webkit.org/show_bug.cgi?id=241793

Reviewed by Simon Fraser.

Extent tracking is not used for remote display list recording. It is
currently used for in-memory display list repainting, but the only thing
using that is the GlyphDisplayListCache, and it's not important there.

* Source/WebCore/platform/graphics/DecomposedGlyphs.cpp:
(WebCore::DecomposedGlyphs::create):
(WebCore::DecomposedGlyphs::DecomposedGlyphs):
* Source/WebCore/platform/graphics/DecomposedGlyphs.h:
(WebCore::DecomposedGlyphs::positionedGlyphs const):
(WebCore::DecomposedGlyphs::bounds const): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayList.cpp:
(WebCore::DisplayList::DisplayList::setTracksDrawingItemExtents): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayList.h:
(WebCore::DisplayList::DisplayList::resourceHeap const):
(WebCore::DisplayList::DisplayList::tracksDrawingItemExtents const): Deleted.
(WebCore::DisplayList::DisplayList::addDrawingItemExtent): Deleted.
(): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.cpp:
(WebCore::DisplayList::DrawGlyphs::DrawGlyphs):
(WebCore::DisplayList::m_bounds): Deleted.
(WebCore::DisplayList::DrawLine::localBounds const): Deleted.
(WebCore::DisplayList::DrawLinesForText::localBounds const): Deleted.
(WebCore::DisplayList::DrawDotsForDocumentMarker::localBounds const): Deleted.
(WebCore::DisplayList::DrawFocusRingPath::localBounds const): Deleted.
(WebCore::DisplayList::DrawFocusRingRects::localBounds const): Deleted.
(WebCore::DisplayList::StrokeRect::localBounds const): Deleted.
(WebCore::DisplayList::StrokePath::localBounds const): Deleted.
(WebCore::DisplayList::StrokeEllipse::localBounds const): Deleted.
(WebCore::DisplayList::StrokeLine::localBounds const): Deleted.
(WebCore::DisplayList::StrokeArc::localBounds const): Deleted.
(WebCore::DisplayList::StrokeQuadCurve::localBounds const): Deleted.
(WebCore::DisplayList::StrokeBezierCurve::localBounds const): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
(WebCore::DisplayList::DrawGlyphs::encode const):
(WebCore::DisplayList::DrawGlyphs::decode):
(WebCore::DisplayList::DrawDecomposedGlyphs::DrawDecomposedGlyphs):
(WebCore::DisplayList::DrawFilteredImageBuffer::globalBounds const): Deleted.
(WebCore::DisplayList::DrawFilteredImageBuffer::localBounds const): Deleted.
(WebCore::DisplayList::DrawGlyphs::globalBounds const): Deleted.
(WebCore::DisplayList::DrawGlyphs::localBounds const): Deleted.
(WebCore::DisplayList::DrawDecomposedGlyphs::globalBounds const): Deleted.
(WebCore::DisplayList::DrawDecomposedGlyphs::localBounds const): Deleted.
(WebCore::DisplayList::DrawImageBuffer::globalBounds const): Deleted.
(WebCore::DisplayList::DrawImageBuffer::localBounds const): Deleted.
(WebCore::DisplayList::DrawNativeImage::globalBounds const): Deleted.
(WebCore::DisplayList::DrawNativeImage::localBounds const): Deleted.
(WebCore::DisplayList::DrawSystemImage::globalBounds const): Deleted.
(WebCore::DisplayList::DrawSystemImage::localBounds const): Deleted.
(WebCore::DisplayList::DrawPattern::globalBounds const): Deleted.
(WebCore::DisplayList::DrawPattern::localBounds const): Deleted.
(WebCore::DisplayList::BeginTransparencyLayer::localBounds const): Deleted.
(WebCore::DisplayList::BeginTransparencyLayer::globalBounds const): Deleted.
(WebCore::DisplayList::EndTransparencyLayer::localBounds const): Deleted.
(WebCore::DisplayList::EndTransparencyLayer::globalBounds const): Deleted.
(WebCore::DisplayList::DrawRect::globalBounds const): Deleted.
(WebCore::DisplayList::DrawRect::localBounds const): Deleted.
(WebCore::DisplayList::DrawLine::globalBounds const): Deleted.
(WebCore::DisplayList::DrawLinesForText::globalBounds const): Deleted.
(WebCore::DisplayList::DrawDotsForDocumentMarker::globalBounds const): Deleted.
(WebCore::DisplayList::DrawEllipse::globalBounds const): Deleted.
(WebCore::DisplayList::DrawEllipse::localBounds const): Deleted.
(WebCore::DisplayList::DrawPath::globalBounds const): Deleted.
(WebCore::DisplayList::DrawPath::localBounds const): Deleted.
(WebCore::DisplayList::DrawFocusRingPath::globalBounds const): Deleted.
(WebCore::DisplayList::DrawFocusRingRects::globalBounds const): Deleted.
(WebCore::DisplayList::FillRect::globalBounds const): Deleted.
(WebCore::DisplayList::FillRect::localBounds const): Deleted.
(WebCore::DisplayList::FillRectWithColor::globalBounds const): Deleted.
(WebCore::DisplayList::FillRectWithColor::localBounds const): Deleted.
(WebCore::DisplayList::FillRectWithGradient::globalBounds const): Deleted.
(WebCore::DisplayList::FillRectWithGradient::localBounds const): Deleted.
(WebCore::DisplayList::FillCompositedRect::globalBounds const): Deleted.
(WebCore::DisplayList::FillCompositedRect::localBounds const): Deleted.
(WebCore::DisplayList::FillRoundedRect::globalBounds const): Deleted.
(WebCore::DisplayList::FillRoundedRect::localBounds const): Deleted.
(WebCore::DisplayList::FillRectWithRoundedHole::globalBounds const): Deleted.
(WebCore::DisplayList::FillRectWithRoundedHole::localBounds const): Deleted.
(WebCore::DisplayList::FillLine::globalBounds const): Deleted.
(WebCore::DisplayList::FillLine::localBounds const): Deleted.
(WebCore::DisplayList::FillArc::globalBounds const): Deleted.
(WebCore::DisplayList::FillArc::localBounds const): Deleted.
(WebCore::DisplayList::FillQuadCurve::globalBounds const): Deleted.
(WebCore::DisplayList::FillQuadCurve::localBounds const): Deleted.
(WebCore::DisplayList::FillBezierCurve::globalBounds const): Deleted.
(WebCore::DisplayList::FillBezierCurve::localBounds const): Deleted.
(WebCore::DisplayList::FillPath::globalBounds const): Deleted.
(WebCore::DisplayList::FillPath::localBounds const): Deleted.
(WebCore::DisplayList::FillEllipse::globalBounds const): Deleted.
(WebCore::DisplayList::FillEllipse::localBounds const): Deleted.
(WebCore::DisplayList::PaintFrameForMedia::localBounds const): Deleted.
(WebCore::DisplayList::PaintFrameForMedia::globalBounds const): Deleted.
(WebCore::DisplayList::StrokeRect::globalBounds const): Deleted.
(WebCore::DisplayList::StrokeLine::globalBounds const): Deleted.
(WebCore::DisplayList::StrokeArc::globalBounds const): Deleted.
(WebCore::DisplayList::StrokeQuadCurve::globalBounds const): Deleted.
(WebCore::DisplayList::StrokeBezierCurve::globalBounds const): Deleted.
(WebCore::DisplayList::StrokePath::globalBounds const): Deleted.
(WebCore::DisplayList::StrokeEllipse::globalBounds const): Deleted.
(WebCore::DisplayList::ClearRect::globalBounds const): Deleted.
(WebCore::DisplayList::ClearRect::localBounds const): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.cpp:
(WebCore::DisplayList::Recorder::drawGlyphsAndCacheResources):
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.cpp:
(WebCore::DisplayList::RecorderImpl::recordDrawDecomposedGlyphs):
(WebCore::DisplayList::RecorderImpl::extentFromLocalBounds const): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.h:
(WebCore::DisplayList::RecorderImpl::append):
* Source/WebKit/GPUProcess/graphics/RemoteDisplayListRecorder.cpp:
(WebKit::RemoteDisplayListRecorder::drawDecomposedGlyphs):
(WebKit::RemoteDisplayListRecorder::drawDecomposedGlyphsWithQualifiedIdentifiers):
* Source/WebKit/GPUProcess/graphics/RemoteDisplayListRecorder.h:
* Source/WebKit/GPUProcess/graphics/RemoteDisplayListRecorder.messages.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder<DecomposedGlyphs>::encode):
(IPC::ArgumentCoder<DecomposedGlyphs>::decode):
* Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.cpp:
(WebKit::RemoteDisplayListRecorderProxy::recordDrawDecomposedGlyphs):
* LayoutTests/displaylists/extent-includes-shadow-expected.txt: Removed.
* LayoutTests/displaylists/extent-includes-shadow.html: Removed.
* LayoutTests/displaylists/extent-includes-transforms-expected.txt: Removed.
* LayoutTests/displaylists/extent-includes-transforms.html: Removed.
* LayoutTests/displaylists/layer-dispay-list-expected.txt:
* LayoutTests/displaylists/replay-skip-clipped-rect-expected.txt: Removed.
* LayoutTests/displaylists/replay-skip-clipped-rect.html: Removed.
* LayoutTests/fast/text/glyph-display-lists/glyph-display-list-color-expected.txt:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/platform/mac-wk1/displaylists/extent-includes-shadow-expected.txt: Removed.
* LayoutTests/platform/mac-wk1/displaylists/extent-includes-transforms-expected.txt: Removed.

Canonical link: https://commits.webkit.org/252442@main
